src/nwb2bids/_converters/_dataset_converter.py

```diff
@@ -22,6 +23,70 @@ class DatasetConverter(BaseConverter):
         default=None,
     )
 
+    @classmethod
+    @pydantic.validate_call
+    def from_remote_dandiset(
+        cls,
+        dandiset_id: str = pydantic.Field(pattern=r"^\d{6}$"),
+        api_url: str | None = None,
+        token: str | None = None,
+        limit: int | None = None,
```

Comment:
If this is used for testing purposes only, I think we can define a constant at module level and monkey patch this constant during test.